### ISSUE TITLE
Feature request: Extract utils into a shared library

### ISSUE THREAD

**Issue by DaveOrDead** [kinde-oss/kinde-auth-pkce-js#32]
### Prerequisites

- [X] I have searched the repository’s issues and [Kinde community](https://thekindecommunity.slack.com/archives/C04K34HUV9B) to ensure my feature request isn’t a duplicate
- [X] I have read the [contributing guidelines](https://github.com/kinde-oss/.github/blob/489e2ca9c3307c2b2e098a885e22f2239116394a/CONTRIBUTING.md)
- [X] I agree to the terms in the [code of conduct](https://github.com/kinde-oss/.github/blob/489e2ca9c3307c2b2e098a885e22f2239116394a/CODE_OF_CONDUCT.md)

### What is the problem you’re trying to solve?

Many of the utility functions are common across all our JS based SDKs (vanilla JS, Next, Node, React) and there is unnecessary code duplication.  These are helpers for:
* getting permissions
* getting organizations
* getting feature flags
etc

### What solution would you like to see?

A consolidated shared helper library that can be consumed by all our JS libraries

**Comment by DaveOrDead** [comment]
We now have a universal [TS/JS based SDK](https://github.com/kinde-oss/kinde-typescript-sdk) which is designed to be consumed by both browsers and node runtime environments. 

We should update this library to leverage it

**Comment by atifcppprogrammer** [comment]
Hey @DaveOrDead long time no see ! I would love to work on refactoring this using the TypeScript SDK, please create an issue for this so I can let loose.

**Comment by DaveOrDead** [comment]
Awesome stuff! You're welcome to use this issue to track it - I'll assign you to it

**Comment by atifcppprogrammer** [comment]
@DaveOrDead, there is one thing that has always bothered me about this SDK, is that there are no tests for public interface i.e. `src/createKindeClient.ts`, prior to refactoring this using the `@kinde-oss/kinde-typescript-sdk` as a dependency I think it is better to have a robust test suite for this file, that establishes a clear `specification` of its behavior. This will ensure that post migration we don't have any regression. Thoughts ?

**Comment by DaveOrDead** [comment]
@atifcppprogrammer yep that makes sense, I've been meaning to bolster the coverage. The TS SDK would remove quite a lot of the logic I think as a lot of it lives there already, so possibly would end up writing a few tests for code that becomes redundant but that's probably fine.

**Comment by atifcppprogrammer** [comment]
Hi @DaveOrDead apologies for dragging my feet over this issue, was totally swamped with work / life :roll_eyes:. I'm still willing to take a crack at this and will have time this weekend. Just wanted to confirm that Kinde is still pursuing the current direction of making the TypeScript SDK a dependency for this SDK, and that this work is still needed.

**Comment by DanielRivers** [comment]
Can you hold off for the moment @atifcppprogrammer please, looking at something different internally atm.

cc: @DaveOrDead
